syntax = "proto3";

package levels;

import "google/protobuf/empty.proto";
import "google/protobuf/timestamp.proto";

// Enum for LevelEnum
enum LevelEnum {
  A0 = 0;
  A1 = 1;
  A2 = 2;
  B1 = 3;
  B2 = 4;
  C1 = 5;
  C2 = 6;
}

message LevelDTO {
  string lang_level = 1;
  google.protobuf.Timestamp created_at = 2;
  google.protobuf.Timestamp updated_at = 3;
}

message LevelRequest {
  string level_enum = 1;
}

message LevelIdResponse {
  string id = 1; // UUID
}

message LevelList {
  repeated LevelDTO levels = 1;
}

message LevelIdRequest {
    string level_id = 1;  // UUID
}

message LevelUpdateRequest {
    string level_id = 1;  // UUID
    string updated_data = 2;
}

// Service definition
service LevelServiceGRPC {
  rpc create_levels (google.protobuf.Empty) returns (google.protobuf.Empty);
  rpc get_levels (google.protobuf.Empty) returns (LevelList);
  rpc create_level(LevelRequest) returns (google.protobuf.Empty);
  rpc get_level_id_by_name (LevelRequest) returns (LevelIdResponse);
  rpc get_level_by_id(LevelIdRequest) returns (LevelDTO);
  rpc update_level(LevelUpdateRequest) returns (google.protobuf.Empty);
  rpc delete_level(LevelIdRequest) returns (google.protobuf.Empty);
}

